code/modules/maps/swapmaps.dm

```diff
@@ -325,7 +325,9 @@ swapmap
 		x2+=x1-1
 		y2+=y1-1
 		z2+=z1-1
-		world.maxz=max(z2,world.maxz)	// stretch z if necessary
+		while(z2 > world.maxz) // stretch z if necessary
```

Comment:
you are applying different logic to the original code and I am not sure that is your intention or that it is a good idea.